### ISSUE TITLE
DNS resolution problem for services inside pod in 1.1

### ISSUE THREAD

**Issue by rvrignaud** [kubernetes/kubernetes#18171]
Hello,

I just upgraded the nodes of my GKE production cluster. I migrated the nodes from 1.0.7 to 1.1.2.
It seems that a few percentage of my pods have troubles to resolve name of kubernetes services (in the same namespace). It was working perfectly when nodes where running k8s 1.0.7.

In stracing a logstash application in a pod that show the problem (other pods of the same RC are working fine) I saw folling error:

```
[pid 10214] sendto(16, "{\"stream\":\"stderr\",\"@version\":\"1"..., 586, 0, {sa_family=AF_INET6, sin6_port=htons(9300), inet_pton(AF_INET6, "::ffff:10.227.250.47", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = 586
[pid 10214] sendto(16, "{\"stream\":\"stderr\",\"@version\":\"1"..., 608, 0, {sa_family=AF_INET6, sin6_port=htons(9300), inet_pton(AF_INET6, "::ffff:10.227.250.47", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = 608
[pid 10214] sendto(16, "{\"stream\":\"stderr\",\"@version\":\"1"..., 5896, 0, {sa_family=AF_INET6, sin6_port=htons(9300), inet_pton(AF_INET6, "::ffff:10.227.250.47", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = 5896
[pid 10214] sendto(16, "{\"stream\":\"stderr\",\"@version\":\"1"..., 65536, 0, {sa_family=AF_INET6, sin6_port=htons(9300), inet_pton(AF_INET6, "::ffff:10.227.250.47", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = -1 EMSGSIZE (Message too long)
[pid 10214] +++ exited with 0 +++
SocketError: send: name or service not known
           send at org/jruby/ext/socket/RubyUDPSocket.java:318
       register at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-udp-1.0.0/lib/logstash/outputs/udp.rb:24
           call at org/jruby/RubyProc.java:271
         encode at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-json-1.0.1/lib/logstash/codecs/json.rb:57
        receive at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-udp-1.0.0/lib/logstash/outputs/udp.rb:34
         handle at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/outputs/base.rb:88
    output_func at (eval):166
   outputworker at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/pipeline.rb:244
  start_outputs at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.4-java/lib/logstash/pipeline.rb:166
```

I don't know how to interpret the message:
```
 EMSGSIZE (Message too long)
```

I have pods problematic service name resolution in multiple namespaces with different application and language runtime.

However a simple nslookup, inside the faulty logstash pod, get the correct result:
```
nslookup logstash-aggregator
Server:		10.227.240.10
Address:	10.227.240.10#53

Name:	logstash-aggregator.infrastructure.svc.cluster.local
Address: 10.227.250.47
```

On application with a python runtime the error is 
```
gaierror: [Errno -2] Name or service not known
```

For some pods, problem seems to have gone away by it self (application retry with backoff did resolve service name eventually) but for some pod it seems that it never get healthy.

**Comment by zmerlynn** [comment]
cc @thockin

**Comment by rvrignaud** [comment]
I need to precise that I did restart kube-proxy, docker services and eventually a reboot of the node without any change.

**Comment by zmerlynn** [comment]
We ended up resolving this on Slack. The issue is that the decreased timeout of 250ms (down from 1s) on the userland proxy in 1.1 actually caused a DNS outage in this particular customer. @thockin ended up advising @rvrignaud on how to switch over to the beta iptables proxy, at which point I also discovered that the `beta` annotation wasn't available in v1.1, and we're scurrying to get #18187 out.

But all good now. We do need a way to advise people if this happens, and if it happens more, we may want consider adjusting the timeout again.

**Comment by pires** [comment]
@zmerlynn @thockin care to link `how to switch over to the beta iptables proxy`? Asking for GKE clusters 1.1.7.

**Comment by thockin** [comment]
You can use kubectl to annotate each node with `
net.beta.kubernetes.io/proxy-mode=iptables` and then reboot the node or
just restart kube-proxy

**Comment by pires** [comment]
Yes, I could find that info at http://googlecloudplatform.blogspot.pt/2015/11/enhancements-to-Container-Engine-and-Container-Registry.html. Thank you!

**Comment by jayme-github** [comment]
@rvrignaud pointed me to this issue in slack.
We're running debian jessie based php-fpm containers in GKE 1.1.7 and had a hard time with lots of `php_network_getaddresses: getaddrinfo failed: Name or service not known` errors.
We've now switched to iptables proxy mode but the errors persist (less frequent as it seems but that might be because we had less traffic in the cluster during the testing).

We've now switched back to using IP's instead of names. Is there anything we could do or help with to track this issue down?
